anndata/_core/anndata.py

```diff
@@ -512,7 +513,10 @@ def _init_as_actual(
                 if isinstance(X, s_type.value):
                     break
             else:
-                class_names = ", ".join(c.__name__ for c in StorageType.classes())
+                class_names = ", ".join(
+                    f"{c.__module__}.{c.__name__}" for c in StorageType.classes()
```

Comment:
```suggestion
                    repr(c) for c in StorageType.classes()
```

How about this? Then these classes will print in a similar style to `type(X)`

Reply:
Yes, looks also good to me!